src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -131,11 +131,27 @@ private RootCommand SetupCommandLineAPI()
 
         private int AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath, FileInfo configurationsFilePath, ReportFormat reportFormat, FileInfo outputFilePath, bool runTtk, bool verbose, bool printMessageIfNotTemplate = true, IReportWriter writer = null, bool readConfigurationFile = true, ILogger logger = null)
         { 
-            if (logger == null) {
-                logger = CreateLogger(verbose);
+            bool disposeWriter = false;
+            
+            // Check that output file path provided for sarif report
+            if (writer == null && reportFormat == ReportFormat.Sarif && outputFilePath == null)
+            {
+                // We can't use the logger for this error,
+                // because we need to get the writer to create the logger, but this check has to be done before getting the writer:
+                Console.WriteLine("Output file path was not provided.");
+                return 3;
             }
 
-            bool disposeWriter = false;
+            if (writer == null)
+            {
+                writer = GetReportWriter(reportFormat, outputFilePath);
+                disposeWriter = true;
+            }
```

Comment:
You could combine this with the check above if you wanted.

```
if (writer == null)
{
    // if report and filePath aren't valid, return error.
    // otherwise create writer and set disposal.
}
```